mycli/main.py

```diff
@@ -933,7 +933,9 @@ def get_prompt(self, string):
         string = string.replace('\\u', sqlexecute.user or '(none)')
         string = string.replace('\\h', host or '(none)')
         string = string.replace('\\d', sqlexecute.dbname or '(none)')
-        string = string.replace('\\t', sqlexecute.server_type()[0] or 'mycli')
+        string = string.replace(
+            '\\t',
+            sqlexecute.server_info.species and sqlexecute.server_info.species.name or 'mycli')
```

Comment:
We probably want `mysql`, not `mycli` here?

Reply:
well, that's the current implementation: if we don't understand, say 'mycli' (probably because it may be some renamed fork of mysql)

Reply:
I think maybe it was a typo from the beginning :)

Reply:
hmm, I get your point, it's kind of inconsistent: if this spot is reserved for the server, we should not replace it with the client name. Would it be appropriate to spell the generic server also `MySQL`?

Reply:
I think so. You could set "ServerSpecies.Unknown" to map to string "MySQL", and just use `species.name` here, and never even check for None, like this:

```
string = string.replace('\\t', sqlexecute.server_info.species.name)
```

and in sqlexecute, instead of defaulting `server_info` to None, have this:

```
self.server_info = ServerInfo.from_version_string('unknown')
```

Reply:
> instead of defaulting server_info to None, have this

server_info is never really None, the code goes like
```Python
def __init__(self, ...):
    self.server_info = None
    self.connect()  # server_info is always set inside connect
```
and initializing it to anything other than None is just a completely wasteful object creation. I guess the only reason it is created in `__init__` is to appease the linter :)